flows/vision-zero/cris_import/cris_import.py

```diff
@@ -0,0 +1,378 @@
+from prefect.schedules.clocks import CronClock
+from prefect.run_configs import UniversalRun
+
+kv_store = get_key_value("Vision Zero Development")
+kv_dictionary = json.loads(kv_store)
+
+SFTP_ENDPOINT = kv_dictionary["SFTP_ENDPOINT"]
+ZIP_PASSWORD = kv_dictionary["ZIP_PASSWORD"]
+VZ_ETL_LOCATION = kv_dictionary["VZ_ETL_LOCATION"]
+
+RAW_AIRFLOW_CONFIG_JSON = kv_dictionary["RAW_AIRFLOW_CONFIG"]
+RAW_AIRFLOW_CONFIG = json.loads(RAW_AIRFLOW_CONFIG_JSON)
+
+AWS_DEFAULT_REGION = kv_dictionary["AWS_DEFAULT_REGION"]
+AWS_ACCESS_KEY_ID = kv_dictionary["AWS_ACCESS_KEY_ID"]
+AWS_SECRET_ACCESS_KEY = kv_dictionary["AWS_SECRET_ACCESS_KEY"]
+AWS_CSV_ARCHIVE_BUCKET_NAME = kv_dictionary["AWS_CSV_ARCHIVE_BUCKET_NAME"]
+AWS_CSV_ARCHIVE_PATH_PRODUCTION = kv_dictionary["AWS_CSV_ARCHIVE_PATH_PRODUCTION"]
+AWS_CSV_ARCHIVE_PATH_STAGING = kv_dictionary["AWS_CSV_ARCHIVE_PATH_STAGING"]
+
+
+def skip_if_running_handler(obj, old_state, new_state):
+    """
+    State management function to prevent a flow executing if it's already running
+
+    See: https://github.com/PrefectHQ/prefect/discussions/5373#discussioncomment-2054536
+    """
+
+    if new_state.is_running():
+        client = Client()
+        query = """
+            query($flow_id: uuid) {
+              flow_run(
+                where: {_and: [{flow_id: {_eq: $flow_id}},
+                {state: {_eq: "Running"}}]}
+                limit: 1
+                offset: 1
+              ) {
+                name
+                state
+                start_time
+              }
+            }
+        """
+        response = client.graphql(
+            query=query, variables=dict(flow_id=prefect.context.flow_id)
+        )
+        active_flow_runs = response["data"]["flow_run"]
+        if active_flow_runs:
+            logger = prefect.context.get("logger")
+            message = "Skipping this flow run since there are already some flow runs in progress"
+            logger.info(message)
+            return Skipped(
+                message
+            )  # or returned Cancelled state if you prefer this state in this use case
+    return new_state
+
+
+@task(
+    name="Download archive from SFTP Endpoint",
+    slug="get-zips",
+    max_retries=3,
+    retry_delay=datetime.timedelta(minutes=2),
+)
+def download_extract_archives():
+    """
+    Connect to the SFTP endpoint which receives archives from CRIS and
+    download them into a temporary directory.
+
+    Returns path of temporary directory as a string
+    """
+
+    logger = prefect.context.get("logger")
+    logger.info(sys._getframe().f_code.co_name + "()")
+    zip_tmpdir = tempfile.mkdtemp()
+    rsync = sysrsync.run(
+        verbose=True,
+        options=["-a"],
+        source_ssh=SFTP_ENDPOINT,
+        source="/home/txdot/*zip",
+        sync_source_contents=False,
+        destination=zip_tmpdir,
+    )
+    logger.info("Rsync return code: " + str(rsync.returncode))
+    if rsync.returncode != 0:
```

Comment:
This stuck out as another place for a comment. A quick Google search got me to the answer that 0 = success, and I think that is could be argued that it can be inferred from the code as well. Just an observation.

Reply:
I can totally get behind that. I appreciate the suggestion; it didn't occur to me because `0` is default return code for programs written in C or maybe more generally for system software on a POSIX host. Noting that in the code is totally helpful, and I will add that. Thank you!